packages/transactions/src/__tests__/TransferHotspotV1.spec.ts

```diff
@@ -0,0 +1,82 @@
+import proto from '@helium/proto'
+import { TransferHotspotV1, Transaction } from '..'
+import {
+  usersFixture,
+  bobB58,
+  aliceB58,
+} from '../../../../integration_tests/fixtures/users'
+
+Transaction.config({
+  txnFeeMultiplier: 5000,
+  dcPayloadSize: 24,
+  stakingFeeTxnAddGatewayV1: 40 * 100000,
+  stakingFeeTxnAssertLocationV1: 10 * 100000,
+})
+
+const transferHotspotFixture = async () => {
+  const { bob, alice } = await usersFixture()
+
+  return new TransferHotspotV1({
+    gateway: alice.address,
+    buyer: bob.address,
+    seller: alice.address,
+    amountToSeller: 10,
+    buyerNonce: 1,
+  })
+}
+
+test('create a transfer hotspot txn', async () => {
+  const transferHotspot = await transferHotspotFixture()
+  expect(transferHotspot.gateway?.b58).toBe(aliceB58)
+  expect(transferHotspot.buyer?.b58).toBe(bobB58)
+  expect(transferHotspot.seller?.b58).toBe(aliceB58)
+  expect(transferHotspot.amountToSeller).toBe(10)
+  expect(transferHotspot.buyerNonce).toBe(1)
+  expect(transferHotspot.fee).toBe(55000)
+})
+
+describe('serialize and deserialize', () => {
+  it('serializes a transfer hotspot txn', async () => {
+    const transferHotspot = await transferHotspotFixture()
+    expect(transferHotspot.serialize().length).toBeGreaterThan(0)
+  })
+
+  it('serializes to base64 string', async () => {
+    const transferHotspot = await transferHotspotFixture()
+    const transferHotspotString = transferHotspot.toString()
+    // verify that we can decode it back from its serialized string
+    const buf = Buffer.from(transferHotspotString, 'base64')
+    const decoded = proto.helium.blockchain_txn.decode(buf)
+    expect(decoded.transferHotspot?.amountToSeller?.toString()).toBe('10')
+  })
+
+  it('deserializes from a base64 string', async () => {
+    const transferHotspot = await transferHotspotFixture()
+    const transferHotspotString = transferHotspot.toString()
+    const deserialized = TransferHotspotV1.fromString(transferHotspotString)
+    expect(deserialized.gateway?.b58).toBe(transferHotspot.gateway?.b58)
+    expect(deserialized.buyer?.b58).toBe(transferHotspot.buyer?.b58)
+    expect(deserialized.seller?.b58).toBe(transferHotspot.seller?.b58)
+    expect(deserialized.amountToSeller).toBe(transferHotspot.amountToSeller)
+    expect(deserialized.buyerNonce).toBe(transferHotspot.buyerNonce)
+    expect(deserialized.fee).toBe(transferHotspot.fee)
```

Comment:
we should probably add expects for the signatures here too

Reply:
Added